src/Raven.Server/Documents/Sharding/ShardedContext.cs

```diff
@@ -135,5 +173,61 @@ public bool HasClientConfigurationChanged(long clientConfigurationEtag)
             var actual = Hashing.Combine(lastClientConfigurationIndex, _lastClientConfigurationIndex);
             return actual > clientConfigurationEtag;
         }
+
+        public void UpdateIndexes(RawDatabaseRecord record)
```

Comment:
nest all Index related methods under .Indexes property so it will be ShardedContext.Indexes.Update etc
Avoid creating files with thousand of lines that act as a dump of all the methods (like ClusterStateMachine or TestBase)

Reply:
cc @karmeli87

Reply:
done